### PR TITLE
Bump CodeQL version to 2.10.4

### PR DESCRIPTION
**Pull request by edoardopirovano**
Part of the 2.10.4 release process.

### Merge / deployment checklist

- [x] Confirm this change is backwards compatible with existing workflows.
- [x] Confirm the [readme](https://github.com/github/codeql-action/blob/main/README.md) has been updated if necessary.
- [x] Confirm the [changelog](https://github.com/github/codeql-action/blob/main/CHANGELOG.md) has been updated if necessary.

### CODE REVIEW

File: python-setup/install_tools.ps1 [hunk 1]
```diff
@@ -9,5 +9,6 @@ py -2 -m pip install --user 'virtualenv<20.11'
 py -3 -m pip install --user 'virtualenv<20.11'
 
 # poetry 1.0.10 has error (https://github.com/python-poetry/poetry/issues/2711)
-py -3 -m pip install --user poetry!=1.0.10
+# We aren't compatible with poetry 1.2
+py -3 -m pip install --user "poetry!=1.0.10,>=1.1,<1.2"
```
Comment (henrymercer) [inline python-setup/install_tools.ps1:13]: `!=1.0.10` is now redundant.  Could you add a changenote stating that Poetry 1.2 isn't supported?

File: python-setup/install_tools.sh [hunk 2]
```diff
@@ -25,7 +25,8 @@ python3 -m pip install --user 'virtualenv<20.11'
 #     LGTM_PYTHON_SETUP_VERSION=The currently activated Python version 2.7.18 is not supported by the project (^3.5). Trying to find and use a compatible version. Using python3 (3.8.2) 3
 
 # poetry 1.0.10 has error (https://github.com/python-poetry/poetry/issues/2711)
-python3 -m pip install --user poetry!=1.0.10
+# We aren't compatible with poetry 1.2
+python3 -m pip install --user "poetry!=1.0.10,>=1.1,<1.2"
```
Comment (henrymercer) [inline python-setup/install_tools.sh:29]: Ditto here

**Review by henrymercer — APPROVED** [review]
LGTM to merge when you get to the right step in the release process.  Consider separating out the Poetry update if the release process takes longer than expected so we can release this change sooner.

**Comment by edoardopirovano** [comment]
@henrymercer Could I get another review for the change note please?

**Review by henrymercer — APPROVED** [review]

**Pull request merged**